tests/integration/test_utils_integration.py

```diff
@@ -451,15 +448,19 @@ def test_extract_arbitrary_unitary_two_modes_vectorized(
         )
 
         if isinstance(U, tf.Tensor):
-            final_state = tf.Session().run(
-                tf.einsum("ab,b", U, tf.constant(initial_state.reshape([-1])))
-            )
-            final_state = np.tile(final_state, bsize)
-        else:
-            final_state = U @ initial_state.reshape([-1])
+            U = U.numpy()
 
-        expected_state = eng_ref.run([p0, prog]).state.ket().reshape([-1])
-        assert np.allclose(final_state, expected_state, atol=tol, rtol=0)
+        final_state = U @ initial_state.reshape([-1])
+        expected_state = eng_ref.run([p0, prog]).state.ket()
+        
+        if isinstance(expected_state, tf.Tensor):
+            expected_state = expected_state.numpy()
+
+        if bsize >= 2:
+            for exp_state in expected_state:
+                assert np.allclose(final_state, exp_state.reshape([-1]), atol=tol, rtol=0)
```

Comment:
I seem to remember that, since the tests just broadcast arrays over the batch axis if `bsize>1`, then a call to `np.allclose` still worked. How come this specific test needed an adjustment, but the same pattern elsewhere in the codebase works without change?

Reply:
All the other tests don't have a `batch_size`.
The problem is that reshape([-1]) destroys the batch dimension, so if it's there we need the for loop.

Reply:
Was this test working previously without the for loop? :thinking:

Reply:
For some reason now it works. Mystery.